media-gfx/gnofract4d/gnofract4d-4.3_p20200821.ebuild

```diff
@@ -0,0 +1,55 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7,8} )
+DISTUTILS_SINGLE_IMPL=1
+
+inherit distutils-r1 eutils virtualx xdg
+
+COMMIT="47752e0d240d9f2686c2a69d813ac5112bda1ad3"
+
+DESCRIPTION="A program for drawing beautiful mathematically-based images known as fractals"
+HOMEPAGE="https://fract4d.github.io/gnofract4d/"
+SRC_URI="https://github.com/fract4d/gnofract4d/archive/${COMMIT}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+COMMON_DEPEND="
+	media-libs/libpng:0=
+	virtual/jpeg:0"
+RDEPEND="${COMMON_DEPEND}
+	$(python_gen_cond_dep '
+		dev-python/pycairo[${PYTHON_USEDEP}]
+		dev-python/pygobject:3[${PYTHON_USEDEP}]
+	')
+	x11-libs/gtk+:3[introspection]"
+BDEPEND="virtual/pkgconfig"
+DEPEND="${COMMON_DEPEND}"
+
+distutils_enable_tests pytest
+
+S="${WORKDIR}/${PN}-${COMMIT}"
+
+src_prepare() {
+	distutils-r1_src_prepare
+	sed -i -e "s:share/doc/gnofract4d/:share/doc/${PF}/:" setup.py || die
+}
+
+python_test() {
+	virtx pytest fract4d fract4d_compiler fract4dgui
```

Comment:
We may be able to avoid invoking virtx repeatedly by taking [this](https://gitweb.gentoo.org/repo/gentoo.git/tree/dev-python/pyopengl/pyopengl-3.1.5.ebuild?id=f5064a1f8bedef6ed63d5c7131dd0c83a0c3a58a#n53) approach, although a custom python_test is unavoidable it seems.

I also think we [need](https://dev.gentoo.org/~mgorny/python-guide/pytest.html?highlight=pytest%20vv#skipping-tests-based-on-markers) -vv and || die (this is complicated by the virtx though?)

Reply:
Tried `virtx distutils-r1_src_test` and that works, updated (also moved src_test to have functions in the order of use).